src/metatrain/experimental/nanopet/modules/structures.py

```diff
@@ -0,0 +1,47 @@
+from typing import List
+
+import torch
+from metatensor.torch.atomistic import System
+
+
+def concatenate_structures(systems: List[System]):
+
+    positions = []
+    centers = []
+    neighbors = []
+    species = []
+    cell_shifts = []
+    cells = []
+    node_counter = 0
+
+    for system in systems:
+        positions.append(system.positions)
+        species.append(system.types)
+
+        assert len(system.known_neighbor_lists()) == 1
+        neighbor_list = system.get_neighbor_list(system.known_neighbor_lists()[0])
```

Comment:
Does this line work in the case of ZBL? I assume the system can have more than one NL here

Reply:
Yes, big blunder, thanks @abmazitov. I will change to >= 1. This is supposed to catch people using the model without a neighbor list

Reply:
I'm not sure we have any guarantee that the first NL in `system.known_neighbor_lists()` will be the one you want. So I still think you should explicitly extract the one you want here instead of taking whatever comes out `system.known_neighbor_lists()[0]`.